### PR TITLE
Fix some code caught by shellcheck

### PR DESCRIPTION
**Pull request by mstreuhofer**

### CODE REVIEW

File: main.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # You are not obligated to bundle the LICENSE file with your b3bp projects as long
 # as you leave these references intact in the header comments of your source files.
 
-# Exit on error. Append || true if you expect an error.
+# Exit on error. Append "|| :" if you expect an error.
```
Comment (gdevenyi) [inline main.sh]: Why : instead of true?

Comment (mstreuhofer) [inline main.sh]: because true is an external command

File: main.sh [hunk 2]
```diff
@@ -75,22 +84,22 @@ function __b3bp_log () {
...
-function info ()      { [ "${LOG_LEVEL:-0}" -ge 6 ] && $(__b3bp_log info "${@}") || true; }
-function debug ()     { [ "${LOG_LEVEL:-0}" -ge 7 ] && $(__b3bp_log debug "${@}") || true; }
+function emergency () {                                $(__b3bp_log emergency "${@}"); exit 1; }
+function alert ()     { [ "${LOG_LEVEL:-0}" -ge 1 ] && $(__b3bp_log alert "${@}"); true; }
+function critical ()  { [ "${LOG_LEVEL:-0}" -ge 2 ] && $(__b3bp_log critical "${@}"); true; }
+function error ()     { [ "${LOG_LEVEL:-0}" -ge 3 ] && $(__b3bp_log error "${@}"); true; }
+function warning ()   { [ "${LOG_LEVEL:-0}" -ge 4 ] && $(__b3bp_log warning "${@}"); true; }
+function notice ()    { [ "${LOG_LEVEL:-0}" -ge 5 ] && $(__b3bp_log notice "${@}"); true; }
+function info ()      { [ "${LOG_LEVEL:-0}" -ge 6 ] && $(__b3bp_log info "${@}"); true; }
+function debug ()     { [ "${LOG_LEVEL:-0}" -ge 7 ] && $(__b3bp_log debug "${@}"); true; }
```
Comment (zbeekman) [inline main.sh]: I'd personally prefer wrapping in braces and `# shellcheck disable=SC2015`... Probably not a deal breaker either way

File: main.sh [hunk 3]
```diff
@@ -53,14 +53,23 @@ function __b3bp_log () {
...
+  # shellcheck disable=SC2034
   local color_warning="\x1b[33m"
+  # shellcheck disable=SC2034
   local color_error="\x1b[31m"
+  # shellcheck disable=SC2034
   local color_critical="\x1b[1;31m"
+  # shellcheck disable=SC2034
   local color_alert="\x1b[1;33;41m"
+  # shellcheck disable=SC2034
   local color_emergency="\x1b[1;4;5;33;41m"
```
Comment (zbeekman) [inline main.sh]: Let's wrap this in braces to reduce duplication:

```
#shellcheck disable=SC2034
{
...
}
```

File: main.sh [hunk 4]
```diff
@@ -249,23 +262,23 @@ fi
...
 
 # no color mode
-if [ "${arg_n}" = "1" ]; then
+if [ "${arg_n:?}" = "1" ]; then
   NO_COLOR="true"
 fi
 
 # help mode
-if [ "${arg_h}" = "1" ]; then
+if [ "${arg_h:?}" = "1" ]; then
```
Comment (zbeekman) [inline main.sh]: interesting approach... I was going to have just used disable=... but I think I like this more

**Comment by zbeekman** [comment]
So, to me, the only deal breaker is the duplication of disable=SC... but @kvz can feel free to override my opinions

**Comment by kvz** [comment]
I like this. I can live the the duplication because to me it's clearer what happens compared to looking at a trailing `)`. It's more delete-key-friendly to me if it's separate lines and not structures overlapping many. If that makes sense?

**Pull request merged**